internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -0,0 +1,679 @@
+
+	if veleroBsl == nil {
+		logger.V(1).Info("Velero BackupStorageLocation not found")
+		return false, nil
+	}
+
+	if err != nil {
+		logger.Error(err, "Failed to get Velero BackupStorageLocation")
+		return false, err
+	}
+
+	if err := r.Delete(ctx, veleroBsl); err != nil {
+		logger.Error(err, "Failed to delete Velero BackupStorageLocation")
+		return false, err
+	}
+
+	logger.V(1).Info("Velero BackupStorageLocation deleted")
+
+	return false, nil
+}
+
+// removeNaBSLFinalizerUponVeleroBSLDeletion removes the finalizer from NonAdminBackupStorageLocation
+// after confirming the VeleroBackupStorageLocation is deleted
+func (r *NonAdminBackupStorageLocationReconciler) removeNaBSLFinalizerUponVeleroBSLDeletion(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	if !controllerutil.ContainsFinalizer(nabsl, constant.NabslFinalizerName) {
+		logger.V(1).Info("NonAdminBackupStorageLocation finalizer not found")
+		return false, nil
+	}
+
+	controllerutil.RemoveFinalizer(nabsl, constant.NabslFinalizerName)
+	if err := r.Update(ctx, nabsl); err != nil {
+		logger.Error(err, "Failed to remove finalizer")
+		return false, err
+	}
+
+	logger.V(1).Info("NonAdminBackupStorageLocation finalizer removed")
+
+	return false, nil
+}
+
+// initNaBSLCreate initializes creation of the NonAdminBackupStorageLocation object
+func (r *NonAdminBackupStorageLocationReconciler) initNaBSLCreate(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	if nabsl.Status.Phase != constant.EmptyString {
+		logger.V(1).Info("NonAdminBackupStorageLocation Phase already initialized", constant.CurrentPhaseString, nabsl.Status.Phase)
+		return false, nil
+	}
+
+	// Set phase to New
+	if updated := updateNaBSLPhase(&nabsl.Status.Phase, nacv1alpha1.NaBSLPhaseNew); updated {
+		if err := r.Status().Update(ctx, nabsl); err != nil {
+			logger.Error(err, statusBslUpdateError)
+			return false, err
+		}
+		logger.V(1).Info("NonAdminBackupStorageLocation Phase set to New")
+	} else {
+		logger.V(1).Info("NonAdminBackupStorageLocation Phase update skipped", constant.CurrentPhaseString, nabsl.Status.Phase)
+	}
+	return false, nil
+}
+
+// validateNaBSLSpec validates the NonAdminBackupStorageLocation spec
+func (r *NonAdminBackupStorageLocationReconciler) validateNaBSLSpec(ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	// Skip validation if not in New phase
+	if nabsl.Status.Phase != nacv1alpha1.NaBSLPhaseNew {
+		logger.V(1).Info("Skipping validation, not in New phase", constant.CurrentPhaseString, nabsl.Status.Phase)
+		return false, nil
+	}
+
+	err := function.ValidateBslSpec(ctx, r.Client, nabsl)
+	if err != nil {
+		updatedPhase := updateNonAdminBslPhase(&nabsl.Status.Phase, nacv1alpha1.NaBSLPhaseUnavailable)
```

Comment:
can this confuse users?

Velero BSL in unavailable state means it has not been able to connect to bucket, for example

in our case, BSL was not even created if it fails this step

Reply:
I did not want to introduce new phase and keep it 1-1 same as BSL.

This one is unavailable, but with the condition telling why it's not available (next line)

Reply:
Its fine NAC objects can have their own status phases and interpretation, right ?

Reply:
I will change this to be aligned with phases from Backup and Restore, so they are same and user won't be confused:
https://github.com/migtools/oadp-non-admin/blob/25e40d6abd8c49c43a1a89a9f49d6d489d07707e/api/v1alpha1/nonadmin_types.go#L23-L32

Reply:
> Its fine NAC objects can have their own status phases and interpretation, right ?

I have changed current implementation to align with `Backup` and `Restore` phases so now NAC have one set of phases for all objects.